### PR TITLE
lua package for yaml support

### LINKED ISSUE #6544: Support yaml parser in Lua hooks
Include a YAML parsing library in Lua hooks.

## Context
Using lua hooks, and for validation purposes, we want to be able to parse the tables definitions stored under _lakefs_tables/ which are used for Unity Delta Sharing. Those files are in yaml format, and currently, lua hooks do not support a library for parsing yaml files.

### PR DESCRIPTION
**Pull request by nopcoder**
Close #6544

### CODE REVIEW

File: pkg/actions/lua/encoding/yaml/yaml.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+var yamlLibrary = []lua.RegistryFunction{
+	{Name: "marshal", Function: yamlMarshal},
+	{Name: "unmarshal", Function: yamlUnmarshal},
+}
+
+func check(l *lua.State, err error) {
+	if err != nil {
+		lua.Errorf(l, err.Error())
+		panic("unreachable")
```
Comment (arielshaqed) [inline pkg/actions/lua/encoding/yaml/yaml.go:28]: How expensive is this?  And... are we sure we are exception-safe and `defer x.Close()` for all our Closers along the stack?

**Review by arielshaqed — APPROVED** [review]
Neat!

Can you just add a note about how to test this manually?

**Comment by nopcoder** [comment]
> Neat!
> 
> Can you just add a note about how to test this manually?

1. Thanks for the review!
2. Sure - I've used the `lakefs lua run` capability with the following code:

```lua
local yaml = require("encoding/yaml")

content = [[
config:
    endpoint: http://localhost:8080
]]

local doc = yaml.unmarshal(content)
print("endpoint: ".. doc.config.endpoint)

local content2 = yaml.marshal(doc)
print("----")
print(content2)
print("----")
if content == content2 then
    print("success :)")
else
    print("failed!")
end
```

File: pkg/actions/lua/encoding/yaml/yaml.go (hunk 1 above)
Comment (nopcoder) [reply pkg/actions/lua/encoding/yaml/yaml.go:28]: The lua implementation we are using implemented the error handling using raise (panic), as described https://pkg.go.dev/github.com/Shopify/go-lua#Errorf.

Can go over the current provided files and lookup for resource management issues.

**Pull request merged**